base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -645,11 +605,11 @@ where D: Digest + Send + Sync
         .ok_or_else(|| ChainStorageError::OutOfRange)
     }
 
-    fn fetch_mmr_node_count(&self, tree: MmrTree, height: u64) -> Result<u32, ChainStorageError> {
+    fn fetch_mmr_node_count(&self, tree: MmrTree, height: u64) -> Result<u64, ChainStorageError> {
```

Comment:
Based on the type used for the number of leaf nodes in the MutableMmr the count should be a u32.

Reply:
Agreed, tests failed with bincode IO errors when I changed it to u32 - I tried for a fair amount of time to fix it. Will take another look with fresh eyes.

Reply:
The backend implementation could internally use a u64 (until a fix is found), but I think the backend interface should expose a u32.

Reply:
Ok found the issue ... was the deserialization of a usize instead of u32 for `txos_hash_to_index_db` 🤦

Reply:
Changed to a u32 everywhere